software/firmware/europi_script.py

```diff
@@ -32,18 +35,17 @@ def main():  # 3
        handlers that provide this functionality.
     """
 
-    def __init__(self) -> None:
-        self._state_filename = f"saved_state_{self.__class__.__qualname__}.txt"
+    _last_saved = 0
```

Comment:
Why is `last_saved` a class variable?

Reply:
Yes, but as a class variable it will be shared by all instances, and will not be reset when a new instance is created. The nature of this variable means that it will likely be naturally reset a few seconds after a script starts, so I don't think that under current usage patterns this being a class variable will cause a bug. But I do think that it is an incorrect use of a class variable and not something that we want to make an example of. In the future we may want multiple scripts running in parallel, and this would prevent that.

In addition, I do think that scripts calling `super().__init__()` is a good thing, because (a) it is what a well behaved subclass should do, and (b) it will allow us to more easily add functionality that needs initialization to the base class in the future.

Reply:
This is excellent feedback. My intention was to reduce the amount of required steps for script authors when creating a EuroPi script, since we want to try and be as beginner friendly as possible. However, beginners can still start with the one-off main.py script and completely forego the `EuroPiScript` base class, which is arguably intended for script authors with more Python experience.

I'll remove the class variable and add it back in as an instance variable and add `super().__init__()` back into the documentation.